solvebio/resource/object.py

```diff
@@ -128,3 +129,21 @@ def upload_file(cls, local_path, remote_path, vault_name, **kwargs):
                                                                     obj.path))
 
         return obj
+
+    @property
+    def parent(self):
+        """ Returns the parent object """
+        if self['parent_object_id']:
+            response = self._client.get(
+                os.path.join(self.class_url(), self['parent_object_id']), {})
+            return convert_to_solve_object(response, client=self._client)
+
+        return None
+
+    @property
+    def vault(self):
+        """ Returns the vault object """
+        from . import Vault
+        response = self._client.get(
+            os.path.join(Vault.class_url(), str(self['vault_id'])), {})
```

Comment:
urljoin

Reply:
I guess technically you could do Vault.retrieve(id, client=self._client) also

Reply:
oh ya, thats cleaner. will do.